src/System.Buffers/src/System/Buffers/NativeBufferPool.cs

```diff
@@ -21,6 +40,13 @@ public NativeBufferPool(int maxElementsInBuffer)
             _buckets = new NativeBufferBucket<T>[maxBuckets + 1];
         }
 
+        public void Dispose()
+        {
+            foreach (NativeBufferBucket<T> bucket in _buckets)
```

Comment:
Buckets will never be null since it is constructed in the ctor and is never set to null anywhere, even in the dispose call...therefore, this will never throw a null reference exception